src/plugins/mer/mersettings.cpp

```diff
@@ -226,6 +239,37 @@ void MerSettings::read()
         if (!merEmulator)
             continue;
 
+        //! \todo Does not support multiple (different) emulators (not supported at other places anyway).
+
+        // Reading predefined emulators
+        const bool providedByMerSdk = true;
+        PersistentSettingsReader reader;
+        if (reader.load(globalSettingsFileName())) {
+            const QVariantMap data = reader.restoreValues();
+            const int version = data.value(QLatin1String(MER_DEVICE_MODELS_FILE_VERSION_KEY), 0).toInt();
+            if (version >= 1) {
+                const int count = data.value(QLatin1String(MER_DEVICE_MODELS_COUNT_KEY), 0).toInt();
+                for (int i = 0; i < count; ++i) {
+                    const QString key = QString::fromLatin1(MER_DEVICE_MODELS_DATA_KEY) + QString::number(i);
+                    if (!data.contains(key))
+                        continue;
+
+                    const QVariantMap map = data.value(key).toMap();
+                    const QString name
+                            = map.value(QLatin1String(MER_DEVICE_MODEL_NAME)).toString();
+                    const QSize resolution
+                            = map.value(QLatin1String(MER_DEVICE_MODEL_DISPLAY_RESOLUTION)).toRect().size();
+                    const QSize size
+                            = map.value(QLatin1String(MER_DEVICE_MODEL_DISPLAY_SIZE)).toRect().size();
+                    const QString dconf
+                            = map.value(QLatin1String(MER_DEVICE_MODEL_DCONF_DB)).toString();
+                    merEmulator->addDeviceModel(MerEmulatorDeviceModel(name, resolution, size, dconf, providedByMerSdk));
```

Comment:
The "Move reading SDK provided emulator device models to MerSettings" commit should come first and this shouldn't be done here. MerSettings should only enable access to the list of saved device models. MerEmulatorDevice should only need to know its current device model.

So there should be deviceModels()/addDeviceModel()/removeDeviceModel() in MerSettings and deviceModel()/setDeviceModel() in MerEmulatorDevice. MerSettings should save/read the list of deviceModels in save()/read(). MerEmulatorDevice should store the deviceModel().name in toMap() and restore deviceModel using the stored name in fromMap(). After that also the connection to devicesLoaded will not be needed.

Now I noticed we have an incosistency here - the UI to define the device models is per-emulator but we have a global list of device models stored. Registering multiple emulators is broken currently but it is likely to be fixed soon and after that we can have two emulators that do not support the same set of device models (e.g. latest emulator and an older one which lacks some features).

So we should move the UI to define the custom device models under Options > Sailfish OS > General and later, when multiple emulators are supported, we can extend it with e.g. required OS version field to allow filtering.